File: cloud/Server/ECS/Components/Synapse.cs
// Copyright © Spatial Corporation. All rights reserved.

using Spatial.Simulation;

namespace Spatial.Cloud.ECS.Components;

/// <summary>
/// A weighted connection between two neurons.
/// </summary>
/// <param name="From">The pre-synaptic <see cref="Neuron"/>, from which the <see cref="Synapse"/> originates.</param>
/// <param name="To">The post-synaptic <see cref="Neuron"/>, to which the <see cref="Synapse"/> extends.</param>
/// <param name="Strength">The strength of the <see cref="Synapse"/>.</param>
public record struct Synapse(
    uint From,
    uint To,
    double Strength) : IComponent;

File: interface/data/brain/synapses/Synapse.ts
// Copyright © Spatial Corporation. All rights reserved.

import { Schema } from "../..";

/**
 * A weighted connection between two neurons.
 */
export type Synapse = Schema<{
  /**
   * The neuron the {@link Synapse} extends from.
   */
  from: string;

  /**
   * The neuron the {@link Synapse} extends to.
   */
  to: string;

  /**
   * The strength of the {@link Synapse}.
   */
  strength: number;
}>;
